.github/workflows/feature_test.yml

```diff
@@ -34,8 +34,8 @@ jobs:
           pip install pre-commit
       - name: Run pre-commit on changed files
         run: |
-          git fetch --depth=1 origin main
-          pre-commit run --files $(git diff --diff-filter=d --name-only origin/main...HEAD)
+          git fetch origin $GITHUB_HEAD_REF:$GITHUB_HEAD_REF
```

Comment:
Does this do something different from `--depth=1` or is it just a different way of doing the same thing?

Reply:
Unfortunately, this also doesn't work because those environment variables aren't present unless the workflow is triggered by the `pull_request` event. If it were, `$GITHUB_BASE_REF` would be equivalent to `origin/main` on PRs against `main`, and `$GITHUB_HEAD_REF` matches `HEAD`.

Reply:
Shall we just switch the workflow to the pull_request event? I doubt we are frequently trying to check build status for a commit that doesn't have a PR yet, and if we want to it's easy to create a draft PR

Reply:
I think so. The `synchronize` activity on that event will trigger new runs after merging in the base branch. 

We also have the option of adding `workflow_dispatch` to allow manual runs as a backstop.